convert2rhel/backup/files.py

```diff
@@ -0,0 +1,146 @@
+                    diagnosis="Failed to backup %s. Errno: %s, Error: %s" % (self.filepath, err.errno, err.strerror),
+                )
+        else:
+            loggerinst.info("Can't find %s.", self.filepath)
+
+        # Set the enabled value
+        super(RestorableFile, self).enable()
+
+    def restore(self, rollback=True):
+        """Restore a previously backed up file"""
+        if rollback:
+            loggerinst.task("Rollback: Restore %s from backup" % self.filepath)
+        else:
+            loggerinst.info("Restoring %s from backup" % self.filepath)
+
+        backup_filepath = os.path.join(BACKUP_DIR, os.path.basename(self.filepath))
+
+        # We do not have backup or not backed up by this
+        if not self.enabled or not os.path.isfile(backup_filepath):
+            loggerinst.info("%s hasn't been backed up." % self.filepath)
+            return
+
+        try:
+            shutil.copy2(backup_filepath, self.filepath)
+        except (OSError, IOError) as err:
+            # Do not call 'critical' which would halt the program. We are in
+            # a rollback phase now and we want to rollback as much as possible.
+            # IOError for py2 and OSError for py3
+            loggerinst.warning("Error(%s): %s" % (err.errno, err.strerror))
+            return
+
+        if rollback:
+            loggerinst.info("File %s restored." % self.filepath)
+            super(RestorableFile, self).restore()
+        else:
+            loggerinst.debug("File %s restored." % self.filepath)
+            # not setting enabled to false since this is not being rollback
+            # restoring the backed up file for conversion purposes
+
+    # Probably will be deprecated and unusable since using the BackupController
+    # Depends on specific usage of this
+    def remove(self):
+        """Remove restored file from original place, backup isn't removed"""
+        try:
+            os.remove(self.filepath)
+            loggerinst.debug("File %s removed." % self.filepath)
+        except (OSError, IOError):
+            loggerinst.debug("Couldn't remove restored file %s" % self.filepath)
+
+
+class MissingFile(RestorableChange):
+    """
+    File not present before conversion. Could be created during
+    conversion so should be removed in rollback.
+    """
+
+    def __init__(self, filepath):
+        super(MissingFile, self).__init__()
+        self.filepath = filepath
+
+    def enable(self):
+        if self.enabled:
+            return
+
+        if os.path.isfile(self.filepath):
+            loggerinst.debug(
+                "Shouldn't be called, file {filepath} is present before conversion".format(filepath=self.filepath)
+            )
```

Comment:
What does this mean?

Reply:
This can be caused in a situation, when the `rpm -Va` says that the file is missing, but when the path is checked, the file is there. So it shouldn't be marked as missing on the system. I saw this time to time.

Reply:
I think we can reword it somehow, to me anything that says "shouldn't be called" or "shouldn't happen" and still happens from time to time feels like a contradiction. Can we change it to be more descriptive?